domain/secret/state/state.go

```diff
@@ -2900,6 +2875,43 @@ AND    (subject_type_id = $secretAccessorType.unit_type_id AND subject_id IN ($u
 	return revisionResult, nil
 }
 
+// GetSecretRevisionID returns the revision UUID for the specified secret URI and version,
+// or an error satisfying [secreterrors.SecretRevisionNotFound] if the revision is not found.
+func (s *State) GetSecretRevisionID(ctx context.Context, uri *coresecrets.URI, version int) (uuid.UUID, error) {
+	db, err := s.DB()
+	if err != nil {
+		return uuid.UUID{}, errors.Trace(err)
+	}
+
+	input := secretRevision{
+		SecretID: uri.ID,
+		Revision: version,
+	}
+	stmt, err := s.Prepare(`
+SELECT uuid AS &secretRevision.uuid
+FROM   secret_revision
+WHERE  secret_id = $secretRevision.secret_id
+    AND    revision = $secretRevision.revision`, input)
+	if err != nil {
+		return uuid.UUID{}, errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, input).Get(&input)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return fmt.Errorf("%w: %s/%d", secreterrors.SecretRevisionNotFound, uri, version)
+		}
+		return errors.Trace(err)
+	})
+	if err != nil {
+		return uuid.UUID{}, errors.Trace(err)
+	}
+	parsed, err := uuid.UUIDFromString(input.ID)
```

Comment:
If we need it as a UUID this should be up in the service layer. The db just serves up strings